Fix images for objects without name or rewrite

Hlavtox opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | If a product does not have `link _rewrite` or a `name`, it didn't work because of wrong URL. I use image ID as a fallback. Also, changed `isset` to `!empty` and renamed some variables to make it more understandable.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Resolves #27757, Resolves #30058
| How to test?      | Make a product, set name and link_rewrite in ps_product_lang to empty strings, check in FO that no images for that product are working. I managed to somehow reproduce a scenario where language addition failed and there was no data for single language, and whole shop was without images.
| Possible impacts? | No


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27758)
<!-- Reviewable:end -->

Hlavtox committed:
Fix images for objects without name or rewrite, add comments

saulaski commented on src/Adapter/Image/ImageRetriever.php:
```diff
@@ -203,7 +203,7 @@ public function getImage($object, $id_image)
 
             $url = $this->link->$getImageURL(
-                isset($object->link_rewrite) ? $object->link_rewrite : $object->name,
+                !empty($object->link_rewrite) ? $object->link_rewrite : (!empty($object->name) ? $object->name : $id_image),
```
Nothing really wrong here, just a bit overcomplicated to chain ternary operators this way.
Functional cases are usually better suited in a good old `if` structure.

Progi1984 replied:
```suggestion
                $object->link_rewrite ?: $object->name ?: $id_image,
```

saulaski replied:
@Progi1984 Unfortunately chained ternary operators can't handle undefined properties and might crash. Did you mean this ?
```suggestion
                $object->link_rewrite ?? $object->name ?? $id_image
```
This last suggestion can't handle tricky cases either, like empty strings.
IMHO there is no way to replace empty tests with chained operators, because `empty()` can handle both undefined and falsey values.
That's why I suggested if/else structure to handle each functional case only to improve readibility.

Progi1984 replied:
Oh yeah ! I don't think.

@Hlavtox Use @saulaski solution ;)

saulaski replied:
@Hlavtox We mean the code should be easier to read, no fancy syntax required :)
Something like this:
```suggestion
            if (!empty($object->link_rewrite)) {
                $imageName = $object->link_rewrite;
            } elseif (!empty($object->name) {
                $imageName = $object->name;
            } else {
                $imageName = $id_image;
            }
            $url = $this->link->$getImageURL(
                $imageName,
```

Progi1984 reviewed (commented):
I have a small suggestion with the elvis operator :man_singer:

matthieu-rolland reviewed (commented):
waiting for @Progi1984 's feedback to be applied, LGTM otherwise 👍

matks reviewed (changes requested):
I agree with @saulaski too 👍

Hlavtox commented:
@matks @saulaski @Progi1984 I don't understand which solution you want me to implement. :-)

But I guess my original one did work properly?

kpodemski commented:
Hey @Hlavtox 

@saulaski had a great suggestion to make this code more readable. What do you think? let's try to unblock this PR as it is pretty straightforward to test and merge 👍

matks commented:
> Hey @Hlavtox
> 
> @saulaski had a great suggestion to make this code more readable. What do you think? let's try to unblock this PR as it is pretty straightforward to test and merge 👍

Friendly reminder @Hlavtox 😄

Hlavtox commented:
Hello @saulaski, @matks, @Progi1984, @kpodemski

I implemented the simplier solution as you requested, added some comments and renamed some small things to make it more clear. Ready for review 👍

Progi1984 commented:
@Hlavtox Please ping @PrestaShop/prestashop-maintainers

Hlavtox committed:
Fix CS

Hlavtox commented:
@hibatallahAouadni @sallemiines Ping for QA please, it has been more than 40 days 😐

sallemiines commented:
Hello @Hlavtox !

The issue is solved :+1: 

https://user-images.githubusercontent.com/31663369/199719645-0c478d38-49e9-40a2-a6f5-2ed648bf999b.mp4

Thank you ! :rocket:

jolelievre commented:
Thanks @Hlavtox and @sallemiines
